packages/template-retail-react-app/app/hooks/use-multiship.js

```diff
@@ -0,0 +1,878 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+    useShopperBasketsMutation,
+    useShippingMethodsForShipment
+} from '@salesforce/commerce-sdk-react'
+import {usePickupShipment} from '@salesforce/retail-react-app/app/hooks/use-pickup-shipment'
+import {DEFAULT_SHIPMENT_ID} from '@salesforce/retail-react-app/app/constants'
+
+/**
+ * Custom hook to handle multiship functionality for cart items
+ * @param {Object} basket - The current basket object
+ * @returns {Object} Object containing helper functions for multiship management
+ */
+export const useMultiship = (basket) => {
+    const {
+        isCurrentShippingMethodPickup,
+        getDefaultShippingMethodId,
+        getPickupShippingMethodId,
+        getShippingAddressForStore,
+        configureDefaultShipmentIfNeeded
+    } = usePickupShipment(basket)
+
+    const updateItemInBasketMutation = useShopperBasketsMutation('updateItemInBasket')
+    const createShipmentForBasketMutation = useShopperBasketsMutation('createShipmentForBasket')
+    const removeShipmentFromBasketMutation = useShopperBasketsMutation('removeShipmentFromBasket')
+    const updateShippingMethodForShipmentMutation = useShopperBasketsMutation(
+        'updateShippingMethodForShipment'
+    )
+    const updateItemsInBasketMutation = useShopperBasketsMutation('updateItemsInBasket')
+    const updateShipmentForBasketMutation = useShopperBasketsMutation('updateShipmentForBasket')
+
+    // Hook for shipping methods for the main shipment - we'll use this as a fallback
+    //
+    // TODO: Ideally we would not use the shipping methods for the main shipment on all shipments
+    //
+    const {refetch: refetchShippingMethods} = useShippingMethodsForShipment(
+        {
+            parameters: {
+                basketId: basket?.basketId,
+                shipmentId: DEFAULT_SHIPMENT_ID
+            }
+        },
+        {
+            enabled: false // Disable automatic fetching, we'll fetch manually when needed
+        }
+    )
+
+    /**
+     * Assigns default shipping methods to shipments that don't have one
+     * Note: Currently uses the same shipping methods as the main shipment ('me') for all shipments
+     * This is a limitation due to React hooks constraints - ideally each shipment would get its own shipping methods
+     * @returns {Promise<void>} Promise that resolves when all updates are complete
+     */
+    const assignDefaultShippingMethodsToShipments = async () => {
```

Comment:
I'd advise avoid the word "assign" or "configure" if we are doing any CRUD to the server because it does not make it clear what it actually does. 
```suggestion
    const updateDefaultShippingMethodsToShipments = async () => {}
```

Reply:
changed